vortex-array/src/pipeline/bits/vector.rs

```diff
@@ -0,0 +1,256 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright the Vortex contributors
+
+use std::fmt::{Debug, Formatter};
+use std::ops::Not;
+use std::sync::{Arc, LazyLock};
+
+use bitvec::array::BitArray;
+use bitvec::order::Lsb0;
+
+use super::{BitView, BitViewMut};
+use crate::pipeline::{N, N_BITS};
+
+static EMPTY: LazyLock<BitVector> = LazyLock::new(|| BitVector {
+    bits: Arc::new(BitArray::ZERO),
+    true_count: 0,
+});
+
+static FULL: LazyLock<BitVector> = LazyLock::new(|| BitVector {
+    bits: Arc::new(BitArray::ZERO.not()),
+    true_count: N,
+});
+
+/// An owned fixed-size bit vector of length `N` bits, represented as an array of usize words.
+///
+/// Internally, it uses a [`BitArray`] to store the bits, but this crate has some
+/// performance foot-guns in cases where we can lean on better assumptions, and therefore we wrap
+/// it up for use within Vortex.
+/// Owned bit vector for storing boolean selection masks.
+#[derive(Clone)]
+pub struct BitVector {
+    pub(super) bits: Arc<BitArray<[usize; N_BITS], Lsb0>>,
+    pub(super) true_count: usize,
+}
+
+impl Debug for BitVector {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("BitVector")
+            .field("true_count", &self.true_count)
+            //.field("bits", &self.bits.as_raw_slice())
+            .finish()
+    }
+}
+
+impl PartialEq for BitVector {
+    fn eq(&self, other: &Self) -> bool {
+        Arc::ptr_eq(&self.bits, &other.bits)
+            || (self.true_count == other.true_count && self.bits == other.bits)
+    }
+}
+
+impl Eq for BitVector {}
+
+impl BitVector {
+    pub fn empty() -> &'static BitVector {
+        &EMPTY
+    }
+
+    pub fn full() -> &'static BitVector {
+        &FULL
+    }
+
+    pub fn true_until(n: usize) -> Self {
+        assert!(n <= N, "Cannot create a BitVector with more than N bits");
+
+        let mut bits = Arc::new(BitArray::<[usize; N_BITS], Lsb0>::ZERO);
```

Comment:
Shall we rename `N_BITS` to `N_WORDS`? There are always N bits, this constant counts how many words fit into N?